## In this notebook I start a Skyline DIA project, import my protein database and raw data, then export data as a .csv
  * Emma executed the final PECAN run for me with my DIA data, and sent me the resulting .blib file. 
  * I used the following references for this Skyline workflow: [tutorial slidedeck for SRM](https://github.com/RobertsLab/project-pacific.oyster-larvae/blob/master/Skyline-example-files-ETS.sky/slides01.pdf), Emma's [suggested settings](https://www.evernote.com/shard/s242/sh/2d6c487c-7d9b-423d-a7a0-9a6c64199302/82b8812164cd461f), and this [DIA Analysis Tutorial on the Roberts Lab site](https://github.com/sr320/LabDocs/wiki/DIA-data-Analyses).

#### Start a Skyline project: open Skyline (I used version 3.7.0.11317) 

![New-DIA-Skyline-00](../../images/New-DIA-Skyline-00.png?raw=true)

#### Select "Blank Document", and saved. This project is named **2017-06-02_DNR-2016-Outplant.sky** 
#### Open Settings -> Peptide Settings and selected the following: 
#### Add background proteome, aka protein database, to project. This is the geoduck gonad transcriptome fasta file with the PRTC protein pasted to the end of the file, which is located [on Owl, "GeoTranscriptomePRTC.fasta"](http://owl.fish.washington.edu/generosa/Generosa_DNR/GeoTranscriptomePRTC.fasta). Then work through the rest of the tabs with the following settings: 

![New-DIA-Skyline-01](../../images/New-DIA-Skyline-01.png?raw=true)

![New-DIA-Skyline-02](../../images/New-DIA-Skyline-02.png?raw=true)

![New-DIA-Skyline-03](../../images/New-DIA-Skyline-03.png?raw=true)

![New-DIA-Skyline-04](../../images/New-DIA-Skyline-04.png?raw=true)

![New-DIA-Skyline-05](../../images/New-DIA-Skyline-05.png?raw=true)

#### Under the **LIBRARY** tab, opened and selected the .blib file that was generated by Pecan. This file defines the list of proteins that Pecan found to be present in at least one of my geoduck sample files. 

![New-DIA-Skyline-06](../../images/New-DIA-Skyline-06.png?raw=true)

![New-DIA-Skyline-07](../../images/New-DIA-Skyline-07.png?raw=true)

![New-DIA-Skyline-08](../../images/New-DIA-Skyline-09.png?raw=true)

![New-DIA-Skyline-09](../../images/New-DIA-Skyline-03.png?raw=true)

#### After adjusting all Peptide Settings, hit "OK."  Then, opened **SETTINGS -> TRANSITION SETTINGS** and adjusted all tabs as follows:

![New-DIA-Skyline-10](../../images/New-DIA-Skyline-10.png?raw=true)

![New-DIA-Skyline-11](../../images/New-DIA-Skyline-11.png?raw=true)

![New-DIA-Skyline-12](../../images/New-DIA-Skyline-12.png?raw=true)

![New-DIA-Skyline-13](../../images/New-DIA-Skyline-13.png?raw=true)

![New-DIA-Skyline-14](../../images/New-DIA-Skyline-14.png?raw=true)

#### Select "OK". Next step, populate the analyte tree with the protein database.  Import via FILE -> IMPORT -> FASTA and open the GeoTranscriptomePRTC.fasta.

![New-DIA-Skyline-15](../../images/New-DIA-Skyline-15.png?raw=true)

#### The fasta loads, tells you how many proteins were imported, then populates the left side of the screen with the protein "names" (protein ID from my fasta):

![New-DIA-Skyline-16](../../images/New-DIA-Skyline-16.png?raw=true)

![New-DIA-Skyline-17](../../images/New-DIA-Skyline-17.png?raw=true)

![New-DIA-Skyline-18](../../images/New-DIA-Skyline-18.png?raw=true)

#### Now it's time to import my sample files! Since I demultiplexed my .raw files during the conversion step, I need to import the converted .mzML files into Skyline (as opposed to using .raw files). 

![New-DIA-Skyline-19](../../images/New-DIA-Skyline-19.png?raw=true)

![New-DIA-Skyline-20](../../images/New-DIA-Skyline-20.png?raw=true)

#### Watching import progress ... 

![New-DIA-Skyline-21](../../images/New-DIA-Skyline-21.png?raw=true)

![New-DIA-Skyline-22](../../images/New-DIA-Skyline-22.png?raw=true)

#### And it's complete. I've rearranged the windo via VIEW -> RETENTION TIME -> REPLICATE COMPARISON, and VIEW -> PEAK AREAS -> REPLICATE COMPARISON. 
![New-DIA-Skyline-23](../../images/New-DIA-Skyline-23.png?raw=true)

#### According to Skyline 8,061 proteins were detected across all 10 samples, comprising 30,547 peptdies, with 297,239 transitions. I exported the data by going to FILE -> EXPORT -> REPORT, modified the metrics I'd like in my report to include area data for each transition in each sample.  I also selected the "pivot by replicate" box, since I'll want my data in that form to run an NMDS plot. The report is saved on [Owl, "DIA-Report-wide.csv"](http://owl.fish.washington.edu/generosa/Generosa_DNR/DIA-Report-wide.csv)

![New-DIA-Skyline-24](../../images/New-DIA-Skyline-24.png?raw=true)

#### I also exported the report in long form (aka not pivoted by sample). This is in [Owl, "DIA-Report-long.csv"](http://owl.fish.washington.edu/generosa/Generosa_DNR/DIA-Report-long.csv)

![New-DIA-Skyline-25](../../images/New-DIA-Skyline-25.png?raw=true)

---

### During our PECAN run the PRTC standard peptides were not labeled as "heavy", and therefore they aren't correctly included in the .blib file.  To use the PRTC peptides in our DIA analysis for quality control, Emma provided us with a Skyline file loaded with the PRTC peptides.  This project file can be used to view the consistency of the PRTC retention times and abundances. RT likely shifted during the run, and data _can_ be adjusted to account for this shift.  The abundances should be the same in each sample (and definitely between reps), since we loaded the same amount of PRTC standard into all samples. 

#### Download the SkylineDaily [DIA-PRTC-peptides.sky.zip](http://owl.fish.washington.edu/generosa/Generosa_DNR/PRTC%20peptides.sky.zip) project documents from Owl; unzip the folder. Open the .sky file in a SkylineDaily window

#### Import results (mzML files), just like you did previously. 

![New-DIA-Skyline-26](../../images/New-DIA-Skyline-26.png?raw=true)

---

### Check Skyline's ability to automatically pick peaks: 
Randomly selected 1 peptide in 100 separate proteins, and assigned "correct" (1) or "incorrect" (0) peak selection in [this spreadsheet](.../../analyses/DIA/DIA-Error-Checking.xlsx). Error rate for each replicate differed slightly; overall it was 71%.

#### A couple examples of poor-quality, aka 0
![Example-DIA-0](../../images/Example-DIA-0.PNG?raw=true)
![Example2-DIA-0](../../images/Example2-DIA-0.PNG?raw=true)

#### A couple examples of ok-quality, aka 1
![Example-DIA-1](../../images/Example-DIA-1.PNG?raw=true)
![Example2-DIA-1](https://github.com/RobertsLab/Paper-DNR-Geoduck-Proteomics/blob/master/images/Example2-DIA-1.PNG?raw=true)


